nano/node/confirmation_height_processor.cpp

```diff
@@ -118,7 +118,11 @@ void nano::confirmation_height_processor::run (confirmation_height_mode mode_a)
 				else
 				{
 					lock_and_cleanup ();
-					condition.wait (lk);
```

Comment:
Does this condition need to fire on either original_hash having values, as below, or awaiting_processing having values? It seems like at the top of the loop we could have one wait_until, which checks if either is non-empty, and the body determines which to process.

Reply:
It doesn't need to fire on `original_hash` changing that is just the last popped off value of `awaiting_processing`. The condition below is only hit if the confirmation height processor is paused which is only utilised in a couple tests. I have updated it to make that more clear.